src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/ApiParameterDescriptionExtensions.cs

```diff
@@ -11,6 +12,33 @@ namespace Swashbuckle.AspNetCore.SwaggerGen
 {
     public static class ApiParameterDescriptionExtensions
     {
+        private static readonly Type[] RequiredAttributeTypes = new[]
+        {
+            typeof(BindRequiredAttribute),
+            typeof(RequiredAttribute)
+        };
+
+        public static bool IsRequiredParameter(this ApiParameterDescription apiParameter)
+        {
```

Comment:
As defaulting to `true` for path parameters is applicable to controller and non-controller cases, I would consider starting this method with the following one-liner, and removing that check from the rest of the logic. I think this makes the behavior a bit more explicit and simplifies the rest of the code:

```csharp
if (apiParameter.IsFromPath()) return true;
```

Reply:
Agreed!